tox.ini

```diff
@@ -0,0 +1,115 @@
+[tox]
+envlist =
+    py27-plone-{4.3,5.0,5.1},
+    isort,
+    flake8,
+    docs,
+    coverage-report,
+
+skip_missing_interpreters = false
+
+[testenv]
+skip_install = True
+
+basepython:
+    py27: python2.7
+
+extras =
+    test
+
+commands =
+    mkdir -p {toxinidir}/reports/coverage
+    {envbindir}/buildout -c {toxinidir}/{env:BUILDOUT_FILE} buildout:directory={envdir} buildout:develop={toxinidir} bootstrap
+    {envbindir}/buildout -c {toxinidir}/{env:BUILDOUT_FILE} buildout:directory={envdir} buildout:develop={toxinidir}
+    {envbindir}/code-analysis
+    {envbindir}/coverage run {envbindir}/test -v1 --auto-color {posargs}
+
+setenv =
+    COVERAGE_FILE=.coverage.{envname}
+    plone-4.3: BUILDOUT_FILE="plone-4.3.x.cfg"
+    plone-5.0: BUILDOUT_FILE="plone-5.0.x.cfg"
+    plone-5.1: BUILDOUT_FILE="plone-5.1.x.cfg"
+
+deps =
+    pdbpp
+    setuptools==33.1.1
+    zc.buildout==2.8.0
+    coverage
+
+whitelist_externals =
+    mkdir
+
+[testenv:coverage-report]
+basepython = python2.7
+deps = coverage
+
+setenv =
+    COVERAGE_FILE=.coverage
+
+skip_install = true
+
+commands =
+    coverage erase
+    coverage combine
+    coverage report
+    coverage html
+    coverage xml
+
+[testenv:flake8]
+basepython = python2.7
+skip_install = true
+
+deps =
+    flake8
+    flake8-html
+    flake8-coding
+    flake8-debugger
+    flake8-deprecated
+    flake8-isort
+    flake8-pep3101
+    flake8-plone-hasattr
+    flake8-polyfill
+    flake8-print
+    flake8-quotes
+    flake8-string-format
+    flake8-todo
```

Comment:
The official list of flake8 plugins that are expected to work without any whitelisting is maintained in plone.recipe.codeanalysis:

https://github.com/plone/plone.recipe.codeanalysis/blob/master/setup.py#L71-L84